src/main/java/org/folio/service/orders/lines/update/OrderLinePatchOperationService.java

```diff
@@ -27,38 +54,109 @@ public OrderLinePatchOperationService(RestClient restClient, OrderLinePatchOpera
-            .withPathOrderLineRequest(request)
-            .withStoragePoLine(poLine);
-
-          PatchOperationHandler patchOperationHandler = orderLinePatchOperationHandlerResolver.resolve(request.getOperation());
-          patchOperationHandler.handle(orderLineUpdateInstanceHolder, requestContext)
-              .thenCompose(v -> patchOrderLine(orderLineUpdateInstanceHolder, lineId, requestContext))
-              .thenAccept(v -> future.complete(null))
-              .exceptionally(t -> {
-                future.completeExceptionally(t);
-                return null;
-              });
-        }).exceptionally(t -> {
-          future.completeExceptionally(t);
-          return null;
-        });
+      .thenApply(poLine -> {
+        RequestEntry requestEntry = new RequestEntry(INVENTORY_LOOKUP_ENDPOINTS.get(INSTANCE_RECORDS_BY_ID_ENDPOINT)).withId(newInstanceId);
+        return restClient.getAsJsonObject(requestEntry, requestContext)
+          .thenApply(instanceRecord -> updatePoLineWithInstanceRecordInfo(instanceRecord, poLine))
+          .thenCompose(updatedPoLine -> purchaseOrderLineService.saveOrderLine(updatedPoLine, requestContext))
+          .thenAccept(v -> future.complete(null))
+          .exceptionally(t -> {
+            future.completeExceptionally(t);
+            return null;
+          });
+      }).exceptionally(t -> {
+        logger.error("Error when updating retrieving instance record from inventory-storage request to by instanceId {}, poLineId {}", newInstanceId, lineId);
+        future.completeExceptionally(t);
+        return null;
+      });
 
     return future;
   }
 
-  private CompletableFuture<Void> patchOrderLine(OrderLineUpdateInstanceHolder orderLineUpdateInstanceHolder,
-       String lineId, RequestContext requestContext) {
-    if (Objects.nonNull(orderLineUpdateInstanceHolder.getStoragePatchOrderLineRequest())) {
+  private CompletableFuture<Void> sendPatchOrderLineRequest(OrderLineUpdateInstanceHolder orderLineUpdateInstanceHolder,
+                                                            String lineId, RequestContext requestContext) {
+    StoragePatchOrderLineRequest storagePatchOrderLineRequest = orderLineUpdateInstanceHolder.getStoragePatchOrderLineRequest();
+    if (Objects.nonNull(storagePatchOrderLineRequest)) {
       RequestEntry requestEntry = new RequestEntry(BY_ID_ENDPOINT).withId(lineId);
-      return restClient.patch(requestEntry, orderLineUpdateInstanceHolder.getStoragePatchOrderLineRequest(), requestContext);
+      return restClient.patch(requestEntry, storagePatchOrderLineRequest, requestContext);
     }
     return CompletableFuture.completedFuture(null);
   }
+
+  private PoLine updatePoLineWithInstanceRecordInfo(JsonObject lookupObj, PoLine poLine) {
+    Optional.ofNullable(lookupObj.getJsonArray(INSTANCE_PUBLICATION)).orElse(new JsonArray())
+      .stream()
+      .map(JsonObject.class::cast)
+      .findFirst()
+      .ifPresentOrElse(publication -> {
```

Comment:
It's hard to read when we set poLine properties in different places of the same method. I propose to extract here just object publication and update poLine with this publication at the end of method like we doing for other setters

Reply:
Done. Thanks